### ISSUE TITLE
Intermittent test failure: daemonSuite.TestHTTPAPI

### ISSUE THREAD

**Issue by barrettj12** [canonical/pebble#205]
```
START: daemon_test.go:1057: daemonSuite.TestHTTPAPI
START: daemon_test.go:64: daemonSuite.SetUpTest
PASS: daemon_test.go:64: daemonSuite.SetUpTest	0.000s

2023-03-10T03:39:41.573Z [test] HTTP API server listening on ":0".
2023-03-10T03:39:41.573Z [test] Started daemon.
2023-03-10T03:39:41.576Z [test] [::1]:33068 GET /v1/checks 88.7µs 401
daemon_test.go:1088:
    c.Assert(err, IsNil)
... value *net.OpError = &net.OpError{Op:"close", Net:"tcp", Source:net.Addr(nil), Addr:(*net.TCPAddr)(0xc000493710), Err:poll.errNetClosing{}} ("close tcp [::]:34805: use of closed network connection")

START: daemon_test.go:73: daemonSuite.TearDownTest
PASS: daemon_test.go:73: daemonSuite.TearDownTest	0.000s

FAIL: daemon_test.go:1057: daemonSuite.TestHTTPAPI
```

Possibly related to #174. Full logs here: https://github.com/canonical/pebble/actions/runs/4381094611/jobs/7668828710

**Comment by flotter** [comment]
https://github.com/canonical/pebble/pull/253 Should be fixed by this.

**Comment by benhoyt** [comment]
Fixed by #253, closing.
